packages/angular/src/generators/utils/add-vitest.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  addDependenciesToPackageJson,
+  ensurePackage,
+  joinPathFragments,
+  readJson,
+  type Tree,
+} from '@nx/devkit';
+import { analogVitestAngular, nxVersion } from '../../utils/versions';
+
+export type AddVitestOptions = {
+  name: string;
+  projectRoot: string;
+  skipPackageJson: boolean;
+  strict: boolean;
+};
+
+export async function addVitest(
+  tree: Tree,
+  options: AddVitestOptions
+): Promise<void> {
+  if (!options.skipPackageJson) {
+    process.env.npm_config_legacy_peer_deps ??= 'true';
```

Comment:
Why is this needed?

Reply:
😅 no clue! I just naively mimicked the jest generator thinking that it was a general trick to prevent the generator from failing in case the vitest plugin is not compatible with the angular version or something.

Reply:
Done!